crates/rolldown_common/src/types/symbol_ref.rs

```diff
@@ -29,34 +29,26 @@ impl SymbolRef {
 
   /// Not all symbols have flags info, we only care about part of them.
   /// If you want to ensure the flags info exists, use `flags_mut` instead.
-  pub fn flags<'db>(&self, db: &'db dyn GetLocalDb) -> Option<&'db SymbolRefFlags> {
+  pub fn flags<'db, T: GetLocalDb>(&self, db: &'db T) -> Option<&'db SymbolRefFlags> {
```

Comment:
NIT: Previous code is using syntax sugar and perform the this code, not sure why we want to desugar it.

Reply:
https://doc.rust-lang.org/std/keyword.dyn.html, `dyn` is dynamically dispatched, generic param should be more performant.

Reply:
I thought that `&dyn trait` would be handled specially compared to `Box<dyn trait>`, which rustc would chose to use static dispatch for `&dyn trait` if it could be infered. But I can't find materilas to prove this. Guess I thought it wrong. Glad to know that from this conversation.